src/analyses/variable-sensitivity/value_set_pointer_abstract_object.cpp

```diff
@@ -133,6 +142,15 @@ abstract_object_pointert value_set_pointer_abstract_objectt::write_dereference(
   return shared_from_this();
 }
 
+abstract_object_pointert value_set_pointer_abstract_objectt::typecast(
+  const typet &new_type,
+  const abstract_environmentt &environment,
+  const namespacet &ns) const
+{
+  INVARIANT(is_void_pointer(type()), "Only allow pointer casting from void*");
```

Comment:
I see why but can this be left to the implementations of the objects in the set?

Reply:
This code is placeholder at the moment. More to follow.

Reply:
When we cast a value set of pointers, we need to also cast pointers it contains. We do this now.